### PR TITLE
feat: accessors required for resource generation

### PR DESCRIPTION
**Pull request by asthamohta**

### CODE REVIEW

File: accessors/clients/datastream/interface.go [hunk 1]
```diff
@@ -60,3 +64,29 @@ func (c *DatastreamClientImpl) UpdateStream(ctx context.Context, req *datastream
 	}
 
 }
+
+func (c *DatastreamClientImpl) CreateConnectionProfile(ctx context.Context, createRequest *datastreampb.CreateConnectionProfileRequest) (*operation.OperationWrapper[datastreampb.ConnectionProfile], error) {
+	op, err := c.client.CreateConnectionProfile(ctx, createRequest)
```
Comment (Deep1998) [inline accessors/clients/datastream/interface.go:69]: This seems to be slightly different from what we have been doing in other accessors, where the client interface has method signatures exactly equivalent to the accessors. Based on that logic, this would lie inside accessors.
However, these kind of functions would end up overloading the accessor and I think its better suited here how you have done it.

File: accessors/clients/operation/operation.go
Comment (Deep1998) [inline accessors/clients/operation/operation.go:1]: This file and operation_test.go already seem to exist on master under operation/operation_test. Should we remove those files?

File: accessors/datastream/datastream_accessor_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		dsm					datastreamclient.DatastreamClientMock
+		connectionProfile   *datastreampb.ConnectionProfile
+		getProfileErr       error
+		expectedBucketName  string
+		expectedPrefix		string	
+		expectError 		bool
+	}{
+		{
+			name: "basic correct",
+			connectionProfile: &datastreampb.ConnectionProfile{Profile: &datastreampb.ConnectionProfile_GcsProfile{GcsProfile: &datastreampb.GcsProfile{Bucket: "bucket", RootPath: "/"}}},
```
Comment (Deep1998) [inline accessors/datastream/datastream_accessor_test.go:50]: Please add another test case where rootPath is /test-path, /test_path/, "" (empty string).

Comment (Deep1998) [reply accessors/datastream/datastream_accessor_test.go:50]: Whatever is allowed in rootPath basically, it may be possible that rootPath must always start with "/" or something like that in which case we may skip those cases.

File: accessors/datastream/datastream_accessor.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	gcsProfile := res.Profile.(*datastreampb.ConnectionProfile_GcsProfile).GcsProfile
+	bucketName := gcsProfile.Bucket
+	prefix := gcsProfile.RootPath + datastreamDestinationConnCfg.Prefix
+	prefix = utils.ConcatDirectoryPath(prefix, "data/")
+	return bucketName, prefix, nil
+}
+
+// Deletes a connection Profile
+func (da *DatastreamAccessorImpl) DeleteConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, id string, projectId string, region string) error {
+	op, err :=datastreamClient.DeleteConnectionProfile(ctx,  &datastreampb.DeleteConnectionProfileRequest{
```
Comment (Deep1998) [inline accessors/datastream/datastream_accessor.go:63]: nit: looks like formatting is disabled? ":=datastreamclient"

File: accessors/datastream/datastream_accessor.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+// Creates new connection Profile
+func (da *DatastreamAccessorImpl) CreateConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, req *datastreampb.CreateConnectionProfileRequest) (*datastreampb.ConnectionProfile, error) {
+	op, err :=datastreamClient.CreateConnectionProfile(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return op.Wait(ctx)
+}
```
Comment (Deep1998) [inline accessors/datastream/datastream_accessor.go:84]: nit: new line between functions.

File: accessors/datastream/datastream_accessor.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	
+	err = op.Wait(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates new connection Profile
+func (da *DatastreamAccessorImpl) CreateConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, req *datastreampb.CreateConnectionProfileRequest) (*datastreampb.ConnectionProfile, error) {
```
Comment (Deep1998) [inline accessors/datastream/datastream_accessor.go:78]: What we had decided for accessors was that we should avoid request bodies as function parameters. Instead, we create this request inside the function while the parameters just take the required strings/data etc. Like what is done in DataflowAccessorImpl.LaunchDataflowTemplate().

File: accessors/datastream/datastream_accessor.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+func (da *DatastreamAccessorImpl) GetConnProfilesRegion(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, projectId string, region string) ([]string, error) {
+	profilesIt := datastreamClient.ListConnectionProfiles(ctx, &datastreampb.ListConnectionProfilesRequest{Parent: "projects/" + projectId + "/locations/" + region})
+	var profiles []string = []string{}
+	for {
+		resp, err := profilesIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return profiles, err
```
Comment (Deep1998) [inline accessors/datastream/datastream_accessor.go:95]: Wondering if its a good idea to return whatever profiles we have collected till now. Most uses cases I have seen till now seem to return a nil/empty in case of error. Also keeps the output deterministic in case datastream decides to change orders.

File: accessors/spanner/mocks.go [hunk 7]
```diff
@@ -18,18 +18,25 @@ import (
...
-	CreateChangeStreamMock          func(ctx context.Context, adminClient spanneradmin.AdminClient, changeStreamName, dbURI string) error
+	GetDatabaseDialectMock          	func(ctx context.Context, adminClient spanneradmin.AdminClient, dbURI string) (string, error)
+	CheckExistingDbMock             	func(ctx context.Context, adminClient spanneradmin.AdminClient, dbURI string) (bool, error)
+	CreateEmptyDatabaseMock         	func(ctx context.Context, adminClient spanneradmin.AdminClient, dbURI string) error
+	GetSpannerLeaderLocationMock    	func(ctx context.Context, instanceClient spinstanceadmin.InstanceAdminClient, instanceURI string) (string, error)
+	CheckIfChangeStreamExistsMock   	func(ctx context.Context, changeStreamName, dbURI string) (bool, error)
+	ValidateChangeStreamOptionsMock 	func(ctx context.Context, changeStreamName, dbURI string) error
+	CreateChangeStreamMock          	func(ctx context.Context, adminClient spanneradmin.AdminClient, changeStreamName, dbURI string) error
+	CreateDatabaseMock					func(ctx context.Context, adminClient spanneradmin.AdminClient, dbURI string, conv *internal.Conv, driver string, migrationType string) error
```
Comment (Deep1998) [inline accessors/spanner/mocks.go:34]: nit: even here the formatting seems to be off, do you have gofmt installed?

File: accessors/clients/operation/operation.go
Comment (asthamohta) [reply accessors/clients/operation/operation.go:1]: My merge happened before the files were moved to a new repo, thats why I think it happened I have new files. let me move it

File: accessors/clients/datastream/interface.go (hunk 1 above)
Comment (asthamohta) [reply accessors/clients/datastream/interface.go:69]: Yes, I am dealing with the operation result here as well and not just making a call to the backend, so I think this should be more suitable

File: accessors/datastream/datastream_accessor.go (hunk 5 above)
Comment (asthamohta) [reply accessors/datastream/datastream_accessor.go:78]: In this particular case it becomes a bit complicated as is it possible to have multiple types of connection profiles(eg gcs, mysql, etc) and all of them require substantially different parameters. In either case I would need to create a struct to be passed as types may differ which raises the question for this particular case why not pass the request only?

File: accessors/spanner/mocks.go (hunk 7 above)
Comment (asthamohta) [reply accessors/spanner/mocks.go:34]: This is a github issue, the code looks fine on my IDE and will be fine when I merge it

File: accessors/datastream/datastream_accessor_test.go (hunk 2 above)
Comment (asthamohta) [reply accessors/datastream/datastream_accessor_test.go:50]: For concat, a public utils function is used which should be tested independently, adding a all empty string test case though

File: accessors/datastream/datastream_accessor.go (hunk 5 above)
Comment (Deep1998) [reply accessors/datastream/datastream_accessor.go:77]: Fair makes sense in this case, its not a strict rule but we should do our best to adhere to that.

**Review by Deep1998 — APPROVED** [review]
LGTM, one minor comment

File: accessors/datastream/datastream_accessor_test.go (hunk 2 above)
Comment (asthamohta) [reply accessors/datastream/datastream_accessor_test.go]: The root path is based on what is returned by the request itself so no special handling is required
